firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteSchema.java

```diff
@@ -169,63 +174,77 @@ private void removeMutationBatch(String uid, int batchId) {
   }
 
   private void createQueryCache() {
-    // A cache of targets and associated metadata
-    db.execSQL(
-        "CREATE TABLE targets ("
-            + "target_id INTEGER PRIMARY KEY, "
-            + "canonical_id TEXT, "
-            + "snapshot_version_seconds INTEGER, "
-            + "snapshot_version_nanos INTEGER, "
-            + "resume_token BLOB, "
-            + "last_listen_sequence_number INTEGER,"
-            + "target_proto BLOB)");
-
-    db.execSQL("CREATE INDEX query_targets ON targets (canonical_id, target_id)");
-
-    // Global state tracked across all queries, tracked separately
-    db.execSQL(
-        "CREATE TABLE target_globals ("
-            + "highest_target_id INTEGER, "
-            + "highest_listen_sequence_number INTEGER, "
-            + "last_remote_snapshot_version_seconds INTEGER, "
-            + "last_remote_snapshot_version_nanos INTEGER)");
+    if (!tableExists("targets")) {
```

Comment:
Eh, I named it `V1` since it's the same in both cases. `target_globals` existed but wasn't necessarily populated by the migrations. The query cache lazily created it. However, since it was lazily created, all of that code can handle it existing or not existing.